libraries/AP_NavEKF2/AP_NavEKF2_Outputs.cpp

```diff
@@ -322,10 +320,7 @@ bool NavEKF2_core::getLLH(struct Location &loc) const
         // If no origin has been defined for the EKF, then we cannot use its position states so return a raw
         // GPS reading if available and return false
         if ((gps.status() >= AP_DAL_GPS::GPS_OK_FIX_3D)) {
-            const struct Location &gpsloc = gps.location();
-            loc = gpsloc;
-            loc.relative_alt = 0;
-            loc.terrain_alt = 0;
+            loc = gps.location();
```

Comment:
the only possible danger here would be that the GPS location doesn't not return an absolute altitude although I guess we would already be in trouble in this case because the EKF would report that alt as an absolute alt anyway.

Reply:
Yep.  This moves us towards making `.alt` private and forcing users to use the accessor which would mean it wouldn't matter what frame the GPS put in the location it is returning.